components/resources/quizzes/shared/SkillSelections.tsx

```diff
@@ -13,22 +17,37 @@ type SkillSelectionsProps = {
 
 const SkillSelections: React.FC<SkillSelectionsProps> = ({
   selections,
+  selected,
+  setSelected,
+  currentStage,
   onNextClick,
   onBackClick,
   progress,
   title,
   body,
 }) => {
-  const [selected, setSelected] = useState<string[]>([]);
+  const currentSelections = selected[currentStage] || [];
   const handleSelection = (selection: string) => {
-    if (selected.includes(selection)) {
-      setSelected(selected.filter((s) => s !== selection));
+    const selectionIndex = currentSelections.indexOf(selection);
```

Comment:
what happens if a quiz has the same string on two different questions?

Reply:
That works fine!